uwcirg/isacc-messaging-client-sof
# Allow retroactive edits of existing CarePlan

- Allow retroactive edits of existing CarePlan https://www.pivotaltracker.com/story/show/184551787
- Fix incorrect birthday message date/time https://www.pivotaltracker.com/story/show/182907996


### Testing instructions for CarePlan editing
1. Open Enrollment client for a patient that doesn't have a CarePlan yet. You should be taken to the new Enrollment client like normal.
- test patient note still works
- test that adding and removing message list entries still works
- the CarePlan and CommunicationRequest resources that are created (when hitting done) match what you entered. No extraneous resources are created (no “revoked” CommunicationRequests are generated)
- cannot save the plan with empty messages
2. Open Enrollment client for a patient that does have a CarePlan already. When prompted, click “revoke and create new”.
- the old CarePlan is marked “revoked”
- all (not yet executed) CommunicationRequest resources from old CarePlan are marked “revoked”
- CommunicationRequests that were already executed do not change
- Go to 1.
3. Open Enrollment client for a patient that does have a CarePlan already. When prompted, click “edit”.
- patient note shows what the existing CarePlan had. If you change it and save, the existing CarePlan reflects this change (description attribute)
- The message list from the previously created CarePlan is shown, with correct dates/times and content.
- If removing any messages, once you save:
  - the corresponding CommunicationRequest resources are marked “revoked”
  - the corresponding entry in CarePlan.activity has been removed
- Messages that have already been sent (CommunicationRequest resources with status “completed”) are marked as such visually, and cannot be removed
- If changing any message date or text, the corresponding CommunicationRequest resources reflect the changes
